pkg/validator/expression.go

```diff
@@ -386,3 +386,35 @@ func (h expressionSelectorsMatchesAnything) Validate(rule rulefmt.Rule, promethe
 	}
 	return errs
 }
+
+func newExpressionWithNoMetricName(paramsConfig yaml.Node) (Validator, error) {
+	params := struct{}{}
+	if err := paramsConfig.Decode(&params); err != nil {
+		return nil, err
+	}
+	return &expressionWithNoMetricName{}, nil
+}
+
+type expressionWithNoMetricName struct{}
+
+func (e expressionWithNoMetricName) String() string {
+	return "expression with no metric name"
+}
+
+func (e expressionWithNoMetricName) Validate(rule rulefmt.Rule, prometheusClient *prometheus.Client) []error {
+	if prometheusClient == nil {
```

Comment:
nit: If the validator does not use the `prometheusClient` I'm ok just having the signature `(rule rulefmt.Rule, _ *prometheus.Client)` and not checking the `prometheusClient` at all, but it won't make any harm

Reply:
done